dsl.md

```diff
@@ -0,0 +1,130 @@
+
+## Motivation
+
+Serverless computing has gained popularity for its ability to abstract away infrastructure management tasks, enabling developers to focus on application logic. However, orchestrating serverless workflows across multiple environments often involves dealing with disparate tools and platforms, leading to complexity and inefficiency.
+
+Serverless Workflow addresses this challenge by providing a DSL specifically designed for serverless workflow orchestration. By abstracting away the underlying infrastructure complexities and offering a modular and extensible framework, Serverless Workflow aims to streamline the development, deployment, and management of serverless workflows.
+
+## Design
+
+The Serverless Workflow DSL is crafted with a design philosophy that prioritizes clarity, expressiveness, and ease of use. Its foundation lies in linguistic fluency, emphasizing readability and comprehension. By adopting a fluent style, the DSL promotes intuitive understanding through natural language constructs. Verbs are employed in the imperative tense to denote actions, enhancing clarity and directness in expressing workflow logic. This imperative approach empowers developers to articulate their intentions succinctly and effectively.
+
+The DSL also embraces the principle of implicit default behaviors, sparing authors from unnecessary repetition and enhancing the conciseness of workflow definitions. For instance, default settings alleviate the burden of explicitly defining every detail, streamlining the workflow design process. Furthermore, the DSL allows both inline declaration of components or the creation of reusable elements, granting flexibility in workflow composition. This flexibility allows developers to seamlessly integrate inline task definitions without imposing rigid structural requirements.
+
+Moreover, the DSL eschews strong-typed enumerations wherever feasible, fostering extensibility and adaptability across different runtime environments. While maintaining portability is crucial, the DSL prioritizes customization options for extensions and runtimes, enabling tailored implementations to suit diverse use cases. Additionally, the DSL favors universally understood terms over technical jargon, enhancing accessibility and comprehension for a broader audience.
+
+- Embrace linguistic fluency for enhanced readability and understanding.
+- Utilize imperative verbs to convey actions directly and clearly.
+- Employ implicit default behaviors to reduce redundancy and streamline workflow definitions.
+- Enable the declaration and effortless import of shared components by supporting external references
+- Encourage the declaration of components inline for situations where reusability is unnecessary, prioritizing ease of use in such cases.
+- Prioritize flexibility over strong-typed enumerations for enhanced extensibility.
+- Opt for universally understood terms to improve accessibility and comprehension.
+
+## Versioning
+
+## Version
```

Comment:
I think we can move this to `CONTRIBUTIONS.md`. Might be confusing in the dsl docs.

Reply:
Allright, will do!